assembly/src/parsers/ast/mod.rs

```diff
@@ -0,0 +1,339 @@
+#![allow(dead_code)]
+use super::{AssemblyError, Token, TokenStream, Vec};
+use crate::MODULE_PATH_DELIM;
+use vm_core::utils::{collections::BTreeMap, string::String, string::ToString};
+
+mod nodes;
+use nodes::{Instruction, Node, Procedure};
+
+mod context;
+use context::ParserContext;
+
+mod io_ops;
+mod stack_ops;
+mod u32_ops;
+
+#[cfg(test)]
+pub mod tests;
+
+// TYPE ALIASES
+// ================================================================================================
+type ProcMap = BTreeMap<String, Procedure>;
+
+// Represents a parsed program AST.
+// This AST can then be furthur processed to generate MAST.
+// A program has to have a body and no exported procedures.
+#[derive(Debug)]
+pub struct ProgramAst {
+    pub procedures: ProcMap,
+    pub body: Vec<Node>,
+}
+
+// Represents a parsed module AST.
+// A module can only have exported and local procedures, but no body.
+#[derive(Debug)]
+pub struct ModuleAst {
+    pub procedures: ProcMap,
+}
+
+// PROGRAM COMPILER
+// --------------------------------------------------------------------------------------------
```

Comment:
I would change this to `PARSERS` and would also use "=====" as separators.

Reply:
fixed